### DISCUSSION TITLE
Permissible characters in field keys

### DISCUSSION THREAD

**Discussion by e-t-k** [CBIIT/ccdi-federation-api#9] [API Design Discussion]
All of the keys we've been discussing so far in #7 and #4 consist entirely of lowercase letters and underscore.
Shall we make this the standard going forward? (Or perhaps we did already and I missed it.)

Possibilities:
- Allow only "word" characters (A-Z, a-z, 0-9, and _ )
- Allow only lowercase a-z and underscore; but allow AZaz09_ in unharmonized keys.

Since we are currently using period as a hierarchy delimiter, the federation members are already going to have to do a cleanup pass on `unharmonized` keys to remove periods, so further limiting allowed characters seems reasonable.

**Comment by jakenoble** [comment]
In line with issues #7 and #4, St. Jude and the ongoing API development both use lowercase and underscores for field keys. Standardising this makes sense.

**Comment by claymcleod** [comment]
Addressed in #19.
